msprime/ancestry.py

```diff
@@ -1698,7 +1705,38 @@ class DiracCoalescent(ParametricAncestryModel):
 @dataclasses.dataclass
 class SweepGenicSelection(ParametricAncestryModel):
     """
-    .. todo:: Document me
+    This simulates a coalescent history with a single selective
+    sweep that has occurred in the past. The model is one of a
+    a structured coalescent where selective backgrounds are defined as in
+    `Braverman et al. (1995) <https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1206652/>`_
+    The implementation details here follow closely to those in discoal,
+    `Kern and Schrider (2016)
+    <https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5167068/>`_
+
+    .. warning::
+        As the strength of selection (:math:`2Ns`) increases the chance
+        of two coalescent events happening very very close together in
+        time and can lead to msprime errors. should you encounter this
+        decrease :math:`2Ns` if appropriate.
+
+    :param float position: the location of the beneficial allele along the
+        chromosome.
+    :param float start_frequency: population frequency of the benefical
+        allele at the start of the selective sweep. e.g., for a *de novo*
```

Comment:
e.g. -> E.g.

I think it would be better to delete the "A soft selective sweep" sentence here. This information is better presented in the Examples sections.

Reply:
cut this